dask/dataframe/io/parquet/utils.py

```diff
@@ -476,7 +477,12 @@ def _scrub(i, p):
             "/".join(path_parts[l:])
         )  # use '/'.join() instead of _join_path to be consistent with split('/')
 
+    # numeric rather than glob ordering
+    out_list = sorted(out_list, key=natural_sort_key)
+    file_list = sorted(file_list, key=natural_sort_key)
```

Comment:
Shouldn't this go up before line 455 where the `path_parts_list` is created and then the `out_list` should not be sorted at all. The way this is I am a little worried that the `file_list` and the `out_list` will be ordered differently.

Reply:
good point. The changes to this file are all gone now though, based on @rjzamora's suggestion below. Files should be sorted by callers pre-`_analyze_paths`.

Maybe a second sort in `_analyze_paths` + warning-log makes sense, to check that callers aren't making this mistake?